ctapipe/instrument/camera.py

```diff
@@ -517,6 +517,54 @@ def get_shower_coordinates(self, x, y, psi):
         trans = delta_x * -sin_psi + delta_y * cos_psi
 
         return longi, trans
+    
+    def get_pixel_id(self, x, y):
+        '''
+        Return the camera pixel number which contains a given position (x,y) 
+        in the camera frame. The (x,y) coordinates can be arrays (of equal length),
+        for which the methods returns an array of pixel ids. A warning is raised if the position
+        falls outside the camera.
+
+        Parameters
+        ----------
+        x: astropy.units.Quantity (distance) of horizontal position(s) in the camera frame
+        y: astropy.units.Quantity (distance) of vertical position(s) in the camera frame
+
+        Returns
+        -------
+        pixel_ids: Pixel number or array of pixel numbers. Returns -1 if position falls outside camera
+        '''
+        
+        pixel_centers = np.stack([self.pix_x.to_value(u.m),self.pix_y.to_value(u.m)]).T
+        points_searched = np.dstack([x.to_value(u.m),y.to_value(u.m)])
+
+        kdtree = KDTree(pixel_centers)
```

Comment:
I would suggest to build the kdtree in `__init__`, as it is used by more than one method.

Reply:
agreed - right now we always make the assumption that you cannot modify the geometry (many functions cache a result).  It could also go as a `lazyproperty` like the neighbor matrix,  since most of the time you don't need it.

Reply:
put it into lazyproperty in 7114978